src/lib/settings/settings.py

```diff
@@ -137,8 +171,93 @@ def _get_default_metrics_extractor_role_arn(self, account_id: str) -> str:
         return f"arn:aws:iam::{account_id}:role/role-salmon-cross-account-extract-metrics-dev"
 
     def _process_monitoring_groups(self):
-        # TODO: add wildcards replacement for all the resource types (glue, lambda, etc.)
-        pass
+        # Get resource names dict
+        resource_names = self._get_all_resource_names()
+
+        # Replace wildcards for all the resource types (glue, lambda, etc.)
+        for m_env in self._processed_settings[SettingFileNames.MONITORING_GROUPS].get(
+            "monitoring_groups", []
+        ):
+            for m_res in SettingConfigs.RESOURCE_TYPES:
+                self._replace_wildcards(m_env, m_res, resource_names[m_res])
+
+    def _get_all_resource_names(self) -> dict:
+        """Get all resource names for all the monitored account ids.
+        Returns dict in the following format
+            {"glue_jobs": {
+                "monitored_env_name_1": ["job1", ...,  "jobN"],
+                ...
+                "monitored_env_name_N": ["job1", ...,  "jobN"]]
+                },
+            "glue_workflows": {...},
+            ...
+            }"""
+        # Get all monitored accounts
+        monitored_accounts = self.general.get("monitored_environments", [])
+
+        # Initialize an empty dictionary for each resource type
+        resource_names = dict()
```

Comment:
There is a great container called collections.defaultdict which helps to create dict with no initial definition. You can take a look)

Reply:
Done, thanks for the great proposal! :)